src/test/java/org/variantsync/vevos/simulation/variability/sequenceextraction/LongestNonOverlappingSequencesTest.java

```diff
@@ -0,0 +1,94 @@
+package org.variantsync.vevos.simulation.variability.sequenceextraction;
+
+import org.junit.Test;
+import org.variantsync.functjonal.list.NonEmptyList;
+import org.variantsync.vevos.simulation.variability.SPLCommit;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class LongestNonOverlappingSequencesTest {
+
+    @Test
+    public void exampleExtraction() {
+        // For example, if the commits comprise three partially overlapping sequences ([A-B-C-D-E], [X-Y-Z], [A-B-F-G]),
```

Comment:
dont all branches in a repository have a common ancestor somewhen? Test case might be ok as is though.

Reply:
Normally yes, but some other parts of the class filter merge commits and not all commits can be processed by VEVOS Extraction. Thus,  not all commits have a connection to the common ancestor.

Reply:
True!